### ISSUE TITLE
Handling Non SuccessStatusCodes

### ISSUE THREAD

**Issue by ferrydeboer** [Xablu/Xablu.WebApiClient#22]
I like to have the context when the response.EnsureSuccessStatusCodeAsync() returns false.
Now however the code simple returns the default TResult instance: 
```
if (!await response.EnsureSuccessStatusCodeAsync())
  | return default(TResult);

```

This leaves me in the dark about what happened for choosing an effective strategy. What is the common approach here? The only optin I see at the moment is wrapping the HttpHandler. But this smells since this can be used concurrently.

My first idea is that to have a HttpResponseHandler which by default does what the bottom lines do but could be replaced by a personal implementation. Somtimes non ok response for instance could still contain content as well, albeit of different types.

**Comment by mvanbeusekom** [comment]
@ferrydeboer this should be solved when PR #21  is applied. This will introduce a new class (called `RestApiClient`) which will wrap the result, status code en error message into a result class which can be queried to figure out what is going on.

**Comment by ferrydeboer** [comment]
Ah, that's great. I can wait for that.

**Comment by mvanbeusekom** [comment]
@ferrydeboer we just released version 0.13.0 to NuGet. If you need help or have questions please let us know. For now I will close this issue.

**Comment by ferrydeboer** [comment]
Awesome, I just saw the update coming by on NuGet. Immediately installed it!
